src/fspec-archive.c

```diff
@@ -61,40 +61,27 @@ main(int argc, char **argv)
     struct archive *a = NULL;
     struct archive_entry *entry = NULL;
     const char *prog;
-    int opt;
 
     prog = argc ? strrchr(argv[0], '/') : NULL;
     prog = prog ? prog + 1 : argv[0];
+    argc--;
+    argv++;
 
     a = archive_write_new();
     entry = archive_entry_new();
     if (!a || !entry)
         errx(1, "alloc failure");
 
-    while ((opt = getopt(argc, argv, "x:")) != -1) {
-        switch (opt) {
-        case 'x':
-            if (archive_write_set_format_by_name(a, optarg) != ARCHIVE_OK)
-                errx(1, "%s", archive_error_string(a));
-            break;
-        default:
-            usage(prog);
-        }
-    }
-    argc -= optind;
-    argv += optind;
-
-    if (!archive_format(a) && prog) {
-        if (strcmp(prog, "fspec-tar") == 0)
-            archive_write_set_format_pax_restricted(a);
-        else if (strcmp(prog, "fspec-cpio") == 0)
-            archive_write_set_format_cpio_newc(a);
-    }
-    if (!archive_format(a))
-        errx(1, "archive format could not be inferred, and was not specified explicitly");
+    if (strcmp(prog, "fspec-tar") == 0)
```

Comment:
I think we need to keep the `if (prog)` check. It is possible to exec a program with `argv[0] == NULL`.